Clients/GUI/Pages/CurrentInstrument.razor

```diff
@@ -0,0 +1,292 @@
+                var result = await SendTradeRequest(OperationType.Sell);
+                if (result == false)
+                {
+                    errorOccured = true;
+                    errorMessage = "Unable to sell instrument. Try again later.";
+                }
+                else
+                {
+                    await GetCountFromPortfolio();
+                    await GetUserBalance();
+                }
+            }
+            catch (Exception e)
+            {
+                errorOccured = true;
+                errorMessage = e.Message;
+            }
+        }
+    }
+
+    private async Task Buy()
+    {
+        UpdateBalanceVariable();
+        if (currentBalance < lotsToTrade * instrument.Lot * price)
+            Modal.Show<EmptyModal>("You need more money to make a deal");
+        else
+        {
+            try
+            {
+                var result = await SendTradeRequest(OperationType.Buy);
+                if (result == false)
+                {
+                    errorOccured = true;
+                    errorMessage = "Unable to buy instrument. Try again later.";
+                }
+                else
+                {
+                    await GetCountFromPortfolio();
+                    await GetUserBalance();
+                }
+            }
+            catch (Exception e)
+            {
+                errorOccured = true;
+                errorMessage = e.Message;
+            }
+        }
+    }
+
+    private async Task<bool> SendTradeRequest(OperationType operation)
+    {
+        var request = new TradeRequest()
+        {
+            UserId = userBalance.UserId,
+            Broker = BrokerType.TinkoffBroker,
+            Token = tinkoffToken,
+            Operation = operation,
+            Figi = instrument.Figi,
+            Count = lotsToTrade * instrument.Lot,
+            Price = price,
+            Currency = currency
+        };
+
+        return await BrokerManager.Trade(request);
+    }
+
+    private async Task SetToken()
+    {
+        var formModal = Modal.Show<SetTinkoffTokenModal>("Enter Tinkoff token");
+        var modalResult = await formModal.Result;
+        tinkoffToken = (string)modalResult.Data;
+
+        await Storage.SetItemAsync("tinkoffToken", tinkoffToken);
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        errorOccured = false;
+        isInstrumentLoading = true;
+        instrument = new Instrument()
+        {
+            Figi = "default",
+            Currency = Currency.Rub,
+            Name = "Load instrument",
+            Lot = 0,
+            Price = 0,
+            Type = InstrumentType.Currency
+        };
+
+        try
+        {
+            tinkoffToken = await Storage.GetItemAsync<string>("tinkoffToken");
+
+            if (tinkoffToken == null)
+            {
+                await SetToken();
+            }
+            instrument = await Storage.GetItemAsync<Instrument>("instrument");
```

Comment:
Because I have to send new request to get information about instrument. Do we need this? What's bad in this solution?

Reply:
> Because I have to send new request to get information about instrument. Do we need this? What's bad in this solution?

You want to update storage each time we have such requests? I think that having just a page parameter and a page address like "instrument/{figi}" and [Parameter] Figi is a better option, because using storage here is like an overkill

Reply:
It takes value only one time when page initializes